cmd/snap-update-ns/utils.go

```diff
@@ -286,6 +291,63 @@ func secureMkfileAll(name string, perm os.FileMode, uid, gid int) error {
 	return err
 }
 
+func planWritableMimic(dir string) ([]*Change, error) {
+	// We need a place for "safe keeping" of what is present in the original
+	// directory as we are about to attach a tmpfs there, which will hide
+	// everything inside.
+	logger.Debugf("create-writable-mimic %q", dir)
+	safeKeepingDir := filepath.Join("/tmp/.snap/", dir)
+
+	var changes []*Change
+
+	// Bind mount the original directory elsewhere for safe-keeping.
+	changes = append(changes, &Change{
+		Action: Mount, Entry: mount.Entry{
+			// XXX: should we rbind here?
+			Name: dir, Dir: safeKeepingDir, Options: []string{"bind"}},
+	})
+	// Mount tmpfs over the original directory, hiding its contents.
+	changes = append(changes, &Change{
+		Action: Mount, Entry: mount.Entry{Name: "none", Dir: dir, Type: "tmpfs"},
+	})
+	// Iterate over the items in the original directory (nothing is mounted _yet_).
+	entries, err := ioutilReadDir(dir)
```

Comment:
Not a big deal (or a blocker), cause, like you said, nothing is mounted yet and this is just recording changes, but I think it might be easier to read if you did:
```
entries, err := ioutilReadDir(safeKeepingDir)
```

Reply:
Well, we cannot because that's just us planning and not really doing stuff. The safe keeping directory doesn't even exist yet, let alone is bound mounted.